src/app/public/modules/colorpicker/colorpicker-input.directive.ts

```diff
@@ -191,7 +193,10 @@ export class SkyColorpickerInputDirective
       });
 
     this.skyColorpickerInput.updatePickerValues(this.initialColor);
-    this.skyColorpickerInput.backgroundColorForDisplay = this.initialColor;
+
+    if (!this._disabled) {
+      this.skyColorpickerInput.backgroundColorForDisplay = this.initialColor;
+    }
```

Comment:
Would this also need to check for the `true` condition as well and set `this.skyColorpickerInput.backgroundColorForDisplay = '#fff';` ?

Reply:
No, because this is only triggered once on initialization, which is actually after `setDisabledState` logic is triggered the first time. 

So all I'm doing here is making sure the `backgroundColorForDisplay` is not reset to the `initialColor` value if the component starts off as disabled